Java/parkservices/src/main/java/com/amazon/randomcutforest/parkservices/PredictorCorrector.java

```diff
@@ -415,91 +435,89 @@ protected <P extends AnomalyDescriptor> double centeredTransformPass(P result, f
             position += baseDimension;
+            ++observation;
         }
-        return variation;
+        return (observation == 0) ? 0 : variation / observation;
     }
 
-    /**
-     * the following function determines if the difference is significant. the
-     * difference could arise from the floating point operations in the
-     * transformation. Independently small variations determined by the dynamically
-     * configurable setting of ignoreSimilarFactor may not be interest.
-     *
-     * @param significantScore If the score is significant (but it is possible that
-     *                         we ignore anomalies below/above expected and hence
-     *                         this bit can be modified)
-     * @param point            the current point in question (in RCF space)
-     * @param newPoint         the expected point (in RCF space)
-     * @param startPosition    the (most likely) location of the anomaly
-     * @param result           the result to be vended
-     * @return true if the changes are significant (hence an anomaly) and false
-     *         otherwise
-     */
-
-    protected <P extends AnomalyDescriptor> boolean isSignificant(boolean significantScore, float[] point,
-            float[] newPoint, int startPosition, P result) {
-        checkArgument(point.length == newPoint.length, "incorrect invocation");
-        int baseDimensions = result.getDimension() / result.getShingleSize();
+    protected <P extends AnomalyDescriptor> DiVector constructUncertaintyBox(float[] point, int startPosition,
+            P result) {
         TransformMethod method = result.getTransformMethod();
         boolean differenced = (method == TransformMethod.DIFFERENCE)
                 || (method == TransformMethod.NORMALIZE_DIFFERENCE);
         double[] scale = result.getScale();
         double[] shift = result.getShift();
-        // backward compatibility of older models
-        if (scale == null || shift == null) {
-            return true;
+        int baseDimensions = result.getDimension() / result.getShingleSize();
+        double[] gapLow = new double[baseDimensions];
+        double[] gapHigh = new double[baseDimensions];
+        for (int y = 0; y < baseDimensions; y++) {
+            double a = scale[y] * point[startPosition + y];
+            double shiftBase = shift[y];
+            double shiftAmount = 0;
+            if (shiftBase != 0) {
+                shiftAmount += DEFAULT_NORMALIZATION_PRECISION * (scale[y] + Math.abs(shiftBase));
+            }
+            double pathGap = calculatePathDeviation(point, startPosition, y, baseDimension, differenced);
+            double noiseGap = noiseFactor * result.getDeviations()[baseDimension + y];
+            double gap = max(scale[y] * pathGap, noiseGap) + shiftAmount + DEFAULT_NORMALIZATION_PRECISION;
+            gapLow[y] = max(max(ignoreNearExpectedFromBelow[y], ignoreNearExpectedFromBelowByRatio[y] * Math.abs(a)),
```

Comment:
Yes the interpretation is correct. The confusing part has always been high/low setting in the director. High is triggered when the point is high  -- and in this case above is with respect to the expected (and therefore should go with high). then Tested this with https://github.com/aws/random-cut-forest-by-aws/blob/main/Java/examples/src/main/java/com/amazon/randomcutforest/examples/parkservices/LowNoisePeriodic.java 

uncommenting the setting of ignoreNearExpectedFrom..
However that revealed that the settings have now moved to the builder -- and it is quite possible to (and even likely) to have the settings as public operations for ThresholdedRandonCutForest since these settings can be learned and applied externally. We may have even agreed on adding them back -- but added them now.